### DISCUSSION TITLE
Improving the generated SQL for readability

### DISCUSSION THREAD

**Discussion by tchoutri** [tchoutri/pg-entity#58] [Ideas]
I'd like to tweak the underlying SQL so that in logs it appears (for tables named "raptors" and "turbos") like:

```sql
select r0.name, r0.address, t1.enabled, t1.direction
from raptors as r0
inner join turbos as t1
where r0.id = t1.raptor_id
```

The idea would be to automatically alias the tables with their initial and an incrementing number.

Quite modelled after Elixir's Ecto toolkit.

**Comment by tchoutri** [comment]
@tonyalaribe I'd appreciate your input :)

**Comment by tchoutri** [comment]
@divarvel I'd love your input as well!

**Comment by tonyalaribe** [comment]
I've never seen this in any query builder, but it looks really useful. I also do this a lot with hand written sql, so I'm definitely in favour of this. 

Isn't it a lot of effort to implement though? 
And I really wish you a merry Christmas and happy new year too 😁

**Reply by tchoutri** [reply]
Not too sure, you just have to keep a counter, take the first letter, increment the counter, inject the aliases, and prefix the fields. I already double-quote the fields when processing the eDSL, so it's not too much of a hassle
